ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -714,24 +722,34 @@ class Endpoints(
       .fromFunction((_: E \/ A).leftMap(E.run))
       .recover(logException("Source") andThen Error.fromThrowable andThen (-\/(_)))
 
-  private def httpResponse(
-      output: Future[Error \/ SearchResult[Error \/ JsValue]]
-  ): Future[HttpResponse] =
-    output
-      .map {
-        case -\/(e) => httpResponseError(e)
-        case \/-(searchResult) => httpResponse(searchResult)
-      }
+  private def httpResponse[T](output: T)(implicit T: MkHttpResponse[T]): Future[HttpResponse] =
+    T.run(output)
       .recover(Error.fromThrowable andThen (httpResponseError(_)))
 
+  private implicit def sourceStreamSearchResults[A: JsonWriter]
+      : MkHttpResponse[ET[domain.SyncResponse[Source[Error \/ A, NotUsed]]]] =
+    MkHttpResponse { output =>
+      hrSearchResults.run(output.map(_ map (_ map (_ map ((_: A).toJson)))).run)
```

Comment:
![map map map map](https://c.tenor.com/EYPJjOVJOHYAAAAC/inception-deeper.gif)

Reply:
Slightly concerned about this xD